django/db/migrations/autodetector.py

```diff
@@ -525,27 +533,25 @@ def generate_created_models(self):
         )
         for app_label, model_name in all_added_models:
             model_state = self.to_state.models[app_label, model_name]
-            model_opts = self.new_apps.get_model(app_label, model_name)._meta
             # Gather related fields
             related_fields = {}
             primary_key_rel = None
-            for field in model_opts.local_fields:
+            for field_name, field in model_state.fields.items():
                 if field.remote_field:
                     if field.remote_field.model:
                         if field.primary_key:
                             primary_key_rel = field.remote_field.model
                         elif not field.remote_field.parent_link:
-                            related_fields[field.name] = field
+                            related_fields[field_name] = field
                     # through will be none on M2Ms on swapped-out models;
                     # we can treat lack of through as auto_created=True, though.
-                    if (getattr(field.remote_field, "through", None) and
-                            not field.remote_field.through._meta.auto_created):
-                        related_fields[field.name] = field
-            for field in model_opts.local_many_to_many:
-                if field.remote_field.model:
-                    related_fields[field.name] = field
-                if getattr(field.remote_field, "through", None) and not field.remote_field.through._meta.auto_created:
-                    related_fields[field.name] = field
+                    if getattr(field.remote_field, "through", None):
+                        through_app_label, through_model_name = resolve_relation(
+                            field.remote_field.through,
+                            app_label,
+                            model_name,
+                        )
+                        related_fields[field_name] = field
```

Comment:
@David-Wobrock I checked out your branch to build an alternative to #13904 off model states and I noticed both `through_app_label` and `through_model_name` are unused which makes me believe something is off here.

Reply:
Good catch :dart: 
```python
through_app_label, through_model_name = resolve_relation(...)
```
is unused here and in `generate_deleted_models()`. Do we need them? Also, related comments are outdated.

Reply:
Good catch indeed 👍 I don't know how this bit introduced itself in here, I guess missing attention. I removed it in the last commit.
Still, I haven't touched the PR for quite a while now, so I don't recall everything precisely - but it seems it's not necessary and shouldn't be there.

(I also rebased the PR on the latest `master` commit)